src/app/modules/design-patterns/adapter/interfaces/xml-to-json-adapter.ts

```diff
@@ -0,0 +1,24 @@
+import { Observable, throwError } from 'rxjs';
+import { xml2json } from 'xml-js';
+
+import { IAdapter } from './adapter';
+
+export class XMLtoJsonAdapter implements IAdapter {
+  constructor(private readonly xml: string) {
+    this.convert();
+  }
+
+  convert(): string {
```

Comment:
The return type should be an example of JSON

Reply:
I added a TODO